bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthStoreHandlerImpl.java

```diff
@@ -308,6 +367,16 @@ public Set<String> getAllHandlesFromIndex() {
                         logger.info("Unable to deserialize json, reset LAST_USED to now.  json:\n{}", value);
                         return Instant.now();
                     }
+                } else if (DEVICE_CODE_RESPONSE.equals(recordType)) {
+                    try {
+                        return gson.fromJson(value, DeviceCodeResponse.class);
+                    } catch (Exception e) {
```

Comment:
Is it necessary to catch this very generic `Exception` - according to the docs, [fromJson](https://www.javadoc.io/static/com.google.code.gson/gson/2.12.1/com.google.gson/com/google/gson/Gson.html#fromJson(com.google.gson.JsonElement,java.lang.Class)) should throw [JsonSyntaxException](https://www.javadoc.io/static/com.google.code.gson/gson/2.12.1/com.google.gson/com/google/gson/JsonSyntaxException.html) (at least this is what they declared, not sure what else could happen)